cmd/flightctl-api/main.go

```diff
@@ -250,14 +263,50 @@ func main() {
 			}
 		}
 
+		// Start Metrics server
 		go func() {
+			log.Info("Starting Metrics server")
 			metricsServer := instrumentation.NewMetricsServer(log, cfg, collectors...)
 			if err := metricsServer.Run(ctx); err != nil {
-				log.Fatalf("Error running server: %s", err)
+				log.Errorf("Metrics server error: %v", err)
+				serverErrors <- err
 			}
-			cancel()
 		}()
 	}
 
-	<-ctx.Done()
+	// Wait for shutdown signal or server error
+	log.Info("All servers started, waiting for shutdown signal...")
+	select {
+	case <-ctx.Done():
+		if ctx.Err() == context.Canceled {
+			log.Info("Shutdown signal received, initiating graceful shutdown")
+		} else {
+			log.Infof("Context done (%v), initiating graceful shutdown", ctx.Err())
+		}
+	case err := <-serverErrors:
+		log.Errorf("Server error received: %v", err)
+		log.Info("Initiating shutdown due to server error")
```

Comment:
can we combine these.
```suggestion
		log.Errorf("Server error received terminating: %v", err)
```